src/main/flight/gps_rescue.c

```diff
@@ -118,38 +124,49 @@ typedef struct {
 
 #define GPS_RESCUE_MAX_YAW_RATE          180    // deg/sec max yaw rate
 #define GPS_RESCUE_MIN_DESCENT_DIST_M    5      // minimum descent distance
-#define GPS_RESCUE_MAX_ITERM_VELOCITY    1000   // max iterm value for velocity
-#define GPS_RESCUE_MAX_ITERM_THROTTLE    200    // max iterm value for throttle
-#define GPS_RESCUE_MAX_PITCH_RATE        3000   // max change in pitch per second in degrees * 100
-#define GPS_RESCUE_DISARM_THRESHOLD      2.0f   // disarm threshold in G's
+#define GPS_RESCUE_MAX_THROTTLE_ITERM    200    // max iterm value for throttle in degrees * 100
 
 static float rescueThrottle;
 static float rescueYaw;
 float       gpsRescueAngle[ANGLE_INDEX_COUNT] = { 0, 0 };
 bool        magForceDisable = false;
 static bool newGPSData = false;
 static pt2Filter_t throttleDLpf;
-static pt2Filter_t velocityDLpf;
-static pt3Filter_t pitchLpf;
+static pt1Filter_t velocityDLpf;
+static pt3Filter_t velocityUpsampleLpf;
 
 rescueState_s rescueState;
 
 void gpsRescueInit(void)
 {
-    const float sampleTimeS = HZ_TO_INTERVAL(TASK_GPS_RESCUE_RATE_HZ);
-    float cutoffHz, gain;
+    rescueState.sensor.gpsRescueTaskIntervalSeconds = HZ_TO_INTERVAL(TASK_GPS_RESCUE_RATE_HZ);
+
+    // ***** THIS IS A QUICK HACK replacing the constraint ****
+    const float gpsDataIntervalSeconds = getGpsDataIntervalSeconds();
+    // We need a constant value since we cannot measure the interval with adequate precision in the current code
+    // TODO: set the interval from time between GPS samples using GPS timestamp packet, which helps if the GPS itself drops a packet or has erratic intervals
+    rescueState.sensor.gpsDataIntervalSeconds = 0.1f;
+    if (gpsDataIntervalSeconds > 0.4f) {
+        rescueState.sensor.gpsDataIntervalSeconds = 1.0f;
+    } else if (gpsDataIntervalSeconds > 0.15f) {
+        rescueState.sensor.gpsDataIntervalSeconds = 0.2f;
+    }
+    // Assumes that the only possible options are 1, 5 and 10hz, and that we primarily support 10hz
```

Comment:
At initialization you cannot get a reliable value from `getGpsDataIntervalSeconds()` yet, so this whole block can be deleted.

Reply:
Thanks, I will put it in `RESCUE_INITIALIZE`, to happen at the start of a rescue.

Reply:
Hmm that will be no good because we need an interval in GPS rescue.c all the time.
We would, most likely, need a lowpass on the estimate, and lock it when the lowpass of the estimate is within a certain range.

Reply:
OK I've done that now, and tested it.  Works fine.  
`gps_rescue.c` just works with whatever it gets from `gps.c`.
When we have a better way to estimate the interval in gps.c, an update there is all we need.